### PR TITLE
Allow per-job path override for clusterloader2 results dir

### PR DESCRIPTION
**Pull request by chemikadze**
Some jobs may have layout that is placing results below root artifacts dir.

### CODE REVIEW

**Comment by mm4tt** [comment]
Please rebase
/ok-to-test

Could you provide a sample prow config of cl2 job that outputs results below root artifacts dir? 
What is the reason for that? Cannot you change the config to have CL2 results in /artifacts dir?

**Comment by chemikadze** [comment]
> Could you provide a sample prow config of cl2 job that outputs results below root artifacts dir?
> What is the reason for that? Cannot you change the config to have CL2 results in /artifacts dir?

Don't have OSS example for that, but essentially that's a non-kubetest job that runs clusterloader2, providing --report-dir that's under ARTIFACTS to logically isolate cl2 output from anything else. It is possible to modify it to put it to root dir, though it would be neat to keep that grouping and make perfdash to be able to see past results.

One hypotetical extension of this use case I could see is multiple clusterloader2 runs per job (let say with different parameters), with multiple perfDashArtifactsDir per job specified.

**Comment by chemikadze** [comment]
Rebased, thanks!

File: perfdash/metrics-downloader.go [hunk 1]
```diff
@@ -130,10 +130,10 @@ func (g *Downloader) getJobData(wg *sync.WaitGroup, result JobToCategoryData, re
 					for _, artifact := range artifacts {
 						metricsFileName := filepath.Base(artifact)
 						resultCategory := getResultCategory(metricsFileName, filePrefix, categoryLabel, artifacts)
-						testDataResponse, err := g.MetricsBkt.ReadFile(job, buildNumber,
-							fmt.Sprintf("artifacts/%v", metricsFileName))
+						fileName := g.artifactName(tests, metricsFileName)
+						testDataResponse, err := g.MetricsBkt.ReadFile(job, buildNumber, fileName)
 						if err != nil {
-							klog.Infof("Error when reading response Body: %v", err)
+							klog.Errorf("Error when reading response Body for %q: %v", fileName, err)
```
Comment (wojtek-t) [inline perfdash/metrics-downloader.go:136]: IIRC, those aren't necessary errors - please revert to Infof.

**Review by wojtek-t — COMMENTED** [review]
One minor nit - other than that lgtm.

/approve

**Pull request merged**